### PR TITLE
gh-125593: Use colors to highlight error locations in tracebacks from exception group

### PR DESCRIPTION
**Pull request by wrongnull**
* Issue: gh-125593

### CODE REVIEW

File: Lib/test/test_traceback.py [hunk 1]
```diff
@@ -4637,6 +4637,27 @@ def foo():
...
+                e, capture_locals=True
+            )
+
+        actual = "".join(exc.format(colorize=True))
+        red = _colorize.ANSIColors.RED
+        boldr = _colorize.ANSIColors.BOLD_RED
+        reset = _colorize.ANSIColors.RESET
+
+        self.assertIn(f"{red}1 {reset+boldr}/{reset+red} 0{reset}", actual)
+        self.assertIn(f"{red}~~{reset+boldr}^{reset+red}~~{reset}", actual)
```
Comment (Eclips4) [inline Lib/test/test_traceback.py]: We also need to test for the presence of the magenta color, and, more importantly, we need to ensure that the exception group has been colored and that their sub-exception have also been colored.

**Comment by Eclips4** [comment]
@pablogsal @Yhg1s I think this is on the boundary between a bug fix and a feature, so I wanted to ask for your opinion on backporting it to the 3.13 branch. This pull request is pretty trivial, so I would like to consider it a bug fix.

**Comment by pablogsal** [comment]
It's 100% bug, colors should apply to all exceptions, not just normal ones.

**Comment by Yhg1s** [comment]
Yes, treating this as a bug seems correct, given that the traceback is now _partially_ colourised.

**Comment by wrongnull** [comment]
I'm converting it to a draft while I struggle with test debugging.

**Comment by wrongnull** [comment]
We can't just construct a expected string and compare it's to the actual message from traceback due to that fact that on Windows x86 <object at 0x...> can't be constructed using the f"<object at {hex(id(obj))}>" because it has different formatting for pointers...:
```
First differing element 5:
'  | [54 chars]zed_traceback_from_exception_group.<locals>.foo at 0x06176570>'
'  | [54 chars]zed_traceback_from_exception_group.<locals>.foo at 0x6176570>'
```
@pablogsal What do you think? Is the current approach looks good to you or it need to be more generic?

**Comment by pablogsal** [comment]
> We can't just construct a expected string and compare it's to the actual message from traceback due to that fact that on Windows x86 <object at 0x...> can't be constructed using the f"<object at {hex(id(obj))}>" because it has different formatting for pointers...:
> 
> ```
> First differing element 5:
> '  | [54 chars]zed_traceback_from_exception_group.<locals>.foo at 0x06176570>'
> '  | [54 chars]zed_traceback_from_exception_group.<locals>.foo at 0x6176570>'
> ```
> 
> @pablogsal What do you think? Is the current approach looks good to you or it need to be more generic?

You could use the real `foo` object in the f-string so it's formatted directly in the expectation no? That will respect whatever custom formatting is done per platform

**Comment by wrongnull** [comment]
> You could use the real `foo` object in the f-string so it's formatted directly in the expectation no? That will respect whatever custom formatting is done per platform

I can't. Since pointer formatting is implementation-defined behavior, 32-bit MSVC omits leading zero in formatted string

**Review by Eclips4 — APPROVED** [review]
LGTM.
@pablogsal what do you think about the current approach to testing it?

**Review by pablogsal — APPROVED** [review]
LGTM

**Comment by pablogsal** [comment]
Thanks a lot for the fix @wrongnull and thanks for the review @Eclips4 !

**Pull request merged**